### ISSUE TITLE
Low FPS without clear system resource bottlenecking

### ISSUE THREAD

**Issue by pwkerns** [MaFi-Games/Captain-of-Industry-issues#88]
**Describe the issue**
Low framerate dropping to around 4-5 FPS when screen is focused on city or factory components, improves with focus on empty terrain to normal. This FPS drop does not appear to be associated with maxing out system resources (see attached screengrab from task manager illustrating typical performance during one of these slowdowns. And the game never seems to use more than about 2 gb of RAM despite much more being available.

**To Reproduce**
Happening consistently with any playthrough of CoI on my computer, not savefile dependent.

**Expected behavior**
I expect that if the FPS is dropping it would be a result of maxing out my GPU which does not occur.

**Additional context**
Current Beta Release 0.32
My system specs:
Intel i5 6600K CPU @ 3.5 GHz
32 gb DDDR4 RAM
AMD Radeon R9 380 (latest driver) with 4 gb DDR5 VRAM
ASUS Motherboard with Z170-A Skylake Chipset
Dual Monitor Setup one on displayport one on HDMI

Troubleshooting: I've run Memtest86 to test my memory (no errors), I've benchmarked CPU and GPU rendering on Cinebench which give standard scores for my equipment. I've tried several different BIOS settings (XMP, Default, etc) for my RAM. Overclocking my CPU by 20% had no perceptible impact on FPS. I've double checked that the game is set to use my dedicated GPU in Windows Graphics Settings (I also have no integrated graphics so the dedicated GPU is the same as the power-saver GPU). I've reduced textures to FAST, switched to Fullscreen mode, and reduced resolution substantially but this does not appear to make much improvement in FPS. I noticed that FPS drops further when I am placing belts or pipes and the input/output symbols are overlaid on buildings.

**Attachments**
![image](https://user-images.githubusercontent.com/96752577/147710965-909d637f-8ec1-4726-8201-3b47bf14e928.png)

**Comment by MaFi-Games** [comment]
This is strange indeed. Can you check per-core CPU load? Could it be one core doing something bad? Can you check (or share) logs whether there are some errors?

**Comment by pwkerns** [comment]
Here is my CPU per core usage display, happy to share any logs you need just tell me what they are called and where to find them - I couldn't find any error logs in the Captain of Industry folders.

![image](https://user-images.githubusercontent.com/96752577/147829622-e9775156-cbdf-47a1-ab24-f9b441cad565.png)

**Comment by pwkerns** [comment]
After some additional testing I'm fairly confident this is a hardware or software issue on my end - I've previously been able to play Hardspace Shipbreaker (also Unity Engine) at very good FPS and am now seeing an appreciable slowdown similar to the one I'm encountering in Captain of Industry.

**Comment by MaFi-Games** [comment]
Thank you for letting us know. If this ever occurs again feel free to open another report.

**Comment by WarrenCol** [comment]
And now I have that problem  Seams COI maxes out 4 of my 32 cores.  It isn't using my cores very efficiently.  When I exit the game those 4 core usages drop to normal (matching the other cores)  Just started getting really bad today.  Right after a windows update yesterday.  Maybe if I roll back that update..... I will test that possibility.


Yup, That helped.   The latest update to windows (6-28-2025 is when mine installed) was causing issues with frame rate.  Removed the update, restarted and the game runs better now.  Not maxing out any cores now.

**Comment by Amyclas-Melinoe** [comment]
I had a similar issue in a very early stage of a new game where the game just slowed to a crawl. I suspected it was because my trucks were pathing through a forest to reach a distant oil patch.

Working on that hypothesis, I cleared a path through said forest and was able to run the game at x15 speed without problems after that. It doesn't confirm that pathing is indeed the issue but it's a data point.
